config/app/com/microsoft/azure/iotsolutions/uiconfig/webservice/v1/controllers/PackageController.java

```diff
@@ -83,8 +108,13 @@ public PackageController(IStorage storage) {
 
         final String content = new String(Files.readAllBytes(file.getFile().toPath()));
         final String packageType = data.get(PACKAGE_TYPE_PARAM)[0];
+        final String configType = data.get(PACKAGE_CONFIG_TYPE_PARAM)[0];
```

Comment:
do you always expect a length 1 array?

Reply:
This is because HTML dropdowns can be multi-select and Java API when it receives data automatically cast it to Array.

Reply:
add a comment with that info

Reply:
added

Reply:
The explanation is a bit off but the code is correct. The reasoning is that in this API (as opposed to most of our others) we are getting a multipart form/data content type. You can see formData.asFormUrlEncoded() is returning a Map<String, String[]>. This is so that the form data can handle any possible form input types that may come in. In this case it is just a single value for each field, but if we ever included multiple file select for example then it could be more values.

Reply:
yes rephrased from Elvin's comment.